banzai/tests/test_end_to_end.py

```diff
@@ -0,0 +1,136 @@
+import pytest
+import os
+from glob import glob
+from banzai.dbs import populate_bpm_table, create_db, get_session, CalibrationImage
+from banzai.utils import fits_utils
+import argparse
+
+data_root = os.path.join(os.sep, 'archive', 'engineering')
+
+sites = [os.path.basename(site_path) for site_path in glob(os.path.join(data_root, '???'))]
+instruments = [os.path.join(site, os.path.basename(instrument_path)) for site in sites
+               for instrument_path in glob(os.path.join(os.path.join(data_root, site, '*')))]
+
+days_obs = [os.path.join(instrument, os.path.basename(dayobs_path)) for instrument in instruments
+            for dayobs_path in glob(os.path.join(data_root, instrument, '201*'))]
+
+
+def run_end_to_end_tests():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--marker', dest='marker', help='PyTest marker to run')
+    parser.add_argument('--junit-file', dest='junit_file', help='Path to junit xml file with results')
+    parser.add_argument('--code-path', dest='code_path', help='Path to directory with setup.py')
+    args = parser.parse_args()
+    os.chdir(args.code_path)
+    command = 'python setup.py test -a "--durations=0 --junitxml={junit_file} -m {marker}"'
+    os.system(command.format(junit_file=args.junit_file, marker=args.marker))
+
+
+def run_banzai(entry_point):
+    for day_obs in days_obs:
+        raw_path = os.path.join(data_root, day_obs, 'raw')
+        command = '{cmd} --raw-path {raw_path} --fpack --db-address={db_address}'
+        command = command.format(cmd=entry_point, raw_path=raw_path, db_address=os.environ['DB_ADDRESS'])
+        os.system(command)
+
+
+def get_expected_number_of_calibrations(raw_filenames, calibration_type):
+    number_of_stacks_that_should_have_been_created = 0
+    for day_obs in days_obs:
+        raw_filenames_for_this_dayobs = glob(os.path.join(data_root, day_obs, 'raw', raw_filenames))
+        if calibration_type.lower() == 'skyflat':
+            # Group by filter
+            observed_filters = []
+            for raw_filename in raw_filenames_for_this_dayobs:
+                skyflat_hdu = fits_utils.open_fits_file(raw_filename)
+                observed_filters.append(skyflat_hdu[0].header['FILTER'])
+            observed_filters = set(observed_filters)
+            number_of_stacks_that_should_have_been_created += len(observed_filters)
+        else:
+            # Just one calibration per night
```

Comment:
Should we also check binning? Maybe that's for the future...

Reply:
I think that is a future enhancement.